src/main/java/jenkins/plugins/slack/workflow/SlackResponse.java

```diff
@@ -9,19 +11,23 @@
     private static final String THREAD_ID = "ts";
     private static final String CHANNEL = "channel";
 
+    private SlackService slackService;
```

Comment:
I see in the CI build [1] that this is causing an issue because `slackService` is not serializable. Any recommendations on how to deal with this? Maybe we can restructure this so we don't need to attach `slackService` here, or would it make sense to serialize it somehow?

[1] https://ci.jenkins.io/blue/organizations/jenkins/Plugins%2Fslack-plugin/detail/PR-680/1/pipeline

Reply:
hmm, might be difficult to serialize SlackService, as it holds a run object and the slackuseridresolver,  ideally the userIds would already be resolved by then and slackservice would just have simple types,

probably needs some refactoring...

not sure if making it transient would be safe as you could get a slackResponse early in a build and then keep updating it, if jenkins was rebooted part way through the build then you would lose the slackService reference, but possibly you can lookup the run? not sure if you can get the Context from were you are though

Reply:
The step should be serialized so the information needed to call `getService` when `slackService` is null   and than marking `SlackService` transient will do.

So SlackResponse should save a reference to `SlackSendStep`

Reply:
There is no need to serialize `slackuseridresolver` as you can call the descriptor for getting it.

So I do think `slackService` could be serialized or at least create another object that is highly serializable.

Reply:
I spent a few more minutes looking at this, but need to dig in more. I haven't done much Jenkins plugin development before. Any more explicit guidance you can offer would be appreciated, in the meantime.

For instance, do you literally mean that SlackResponse should save a reference to `SlackSendStep` or to a `SlackSendStepExecution`? The execution looks serializable but I don't think the step itself is. Is that right? Given the Step, or the StepExecution, it's still not totally clear to me how to reconstruct the service when the time comes to do `addReaction`.

Reply:
Perhaps the safest thing is to do nothing if slackService is null, basically it does not make sense to handle pipeline shutdown. WDYT @timja ?

So my suggestion is to mark slackService transient and than look at the value whether it is null and if it is null than return from addReaction

Reply:
```suggestion
    private transient SlackService slackService;
```

Reply:
> Perhaps the safest thing is to do nothing if slackService is null, basically it does not make sense to handle pipeline shutdown. WDYT @timja ?
> 
> So my suggestion is to mark slackService transient and than look at the value whether it is null and if it is null than return from addReaction

that's the quickest fix, but will mean it won't work after a restart of jenkins, but could be good enough for now...